lib/datasource/index.ts

```diff
@@ -67,6 +67,9 @@ async function getRegistryReleases(
     }
   }
   const res = await datasource.getReleases({ ...config, registryUrl });
+  if (res?.releases.length) {
+    res.registryUrl ||= registryUrl;
```

Comment:
```suggestion
    res.registryUrl ??= registryUrl;
```

Only assign it `null` or `undefined`, not when `0` or `false` or `''` (empty string)